File: quarkus_orb/src/main/java/com/codecta/qoq/model/Weapon.java
package com.codecta.qoq.model;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(schema = "quarkusorb", name = "WEAPON")
public class Weapon extends ModelObject {
    @SequenceGenerator(
            name = "WeaponSeq",
            sequenceName = "WEAPON_SEQ",
            schema = "quarkusorb",
            allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "WeaponSeq")
    @Id
    @Column(name = "ID", nullable = false)
    private Integer id;

    private String name;
    private Integer damage;

    @OneToMany(mappedBy = "weapon", fetch = FetchType.LAZY)
    private List<Player> players = new ArrayList<>();

    @OneToMany(mappedBy = "weapon", fetch = FetchType.LAZY)
    private List<Monster> monster = new ArrayList<>();

    @OneToOne(mappedBy = "weapon")
    private Dungeon dungeon;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getDamage() {
        return damage;
    }

    public void setDamage(Integer damage) {
        this.damage = damage;
    }

    public List<Player> getPlayers() {
        return players;
    }

    public void setPlayers(List<Player> players) {
        this.players = players;
    }

    public List<Monster> getMonster() {
        return monster;
    }

    public void setMonster(List<Monster> monster) {
        this.monster = monster;
    }

    public Dungeon getDungeon() {
        return dungeon;
    }

    public void setDungeon(Dungeon dungeon) {
        this.dungeon = dungeon;
    }

    @Override
    public Integer getId() { return this.id; }
}

File: orb_of_quarkus/lib/providers/weapon.dart
import 'package:flutter/foundation.dart';

import 'dungeon.dart';
import 'monster.dart';
import 'player.dart';

class Weapon {
  final int id;
  final String name;
  final int damage;
  final List<Player> players;
  final List<Monster> monsters;
  final Dungeon dungeon;

  Weapon({
    @required this.id,
    @required this.name,
    @required this.damage,
    @required this.players,
    @required this.monsters,
    @required this.dungeon,
  });
}

// class Weapons with ChangeNotifier {
//   List<Weapon> _weapons = [];
//   List<Weapon> get weapons {
//     return [..._weapons];
//   }
// }
